packages/offix-datastore/src/storage/Storage.ts

```diff
@@ -1,57 +1,76 @@
 import { PushStream, ObservablePushStream } from "../utils/PushStream";
-import { Model, PersistedModel } from "../models";
 import { PredicateFunction } from "../predicates";
 import { createDefaultStorage } from "./adapters/defaultStorage";
 import { generateId } from "./core";
 
+/**
+ * This interface defines the API that is required
+ * from any Device specific Storage implementation
+ * It defines a CRUD interface that a
+ * Device Specific Implementation must support
+*/
 export interface IStorageAdapter {
-    save(model: PersistedModel): Promise<PersistedModel>;
-    query(modelName: string, predicate?: PredicateFunction): Promise<PersistedModel | PersistedModel[]>;
-    update(model: PersistedModel): Promise<PersistedModel>;
-    remove(model: PersistedModel, predicate?: PredicateFunction): Promise<PersistedModel | PersistedModel[]>;
+    save(storeName: string, input: any): Promise<any>;
+    query(storeName: string, predicate?: PredicateFunction): Promise<any | any[]>;
+    update(storeName: string, input: any, predicate?: PredicateFunction): Promise<any>;
+    remove(storeName: string, predicate?: PredicateFunction): Promise<any | any[]>;
 }
 
+/**
+ * The various change events that can occur on Store
+ *
+ * "ADD" - data was added to the Store
+ * "UPDATE" - data was updated in the Store
+ * "DELETE" - data was deleted from the Store
+ */
+export type EventTypes = "ADD" | "UPDATE" | "DELETE";
+
+/**
+ * StoreChangeEvent is an event emitted whenever
+ * a change has occurred on the local store
+*/
 export interface StoreChangeEvent {
-    operationType: string;
+    // the type of change event that just occurred
```

Comment:
Can we please do JSDoc comments rather than single line comments. Just for consistency sake :)